components/builder-db/src/migration.rs

```diff
@@ -12,98 +12,21 @@
-        self.xact
-            .execute("SET search_path TO public", &[])
-            .map_err(Error::SchemaSwitch)?;
-
-        // We take this lock because there can be one, and only one, migration
-        // running at a time. No parallel changes allowed.
-        self.xact
-            .execute("SELECT pg_advisory_xact_lock($1)", &[&SETUP_LOCK])
-            .map_err(Error::MigrationLock)?;
-
-        let schema_prefix = self.schema_prefix();
-        for shard in self.shards.iter() {
-            let schema_xact = self.xact.transaction().map_err(Error::TransactionCreate)?;
-            schema_xact
-                .execute("SET search_path TO public", &[])
-                .map_err(Error::SchemaSwitch)?;
-            let schema_name = format!("{}_{}", schema_prefix, shard);
-            let sql_create_schema = format!("CREATE SCHEMA IF NOT EXISTS {}", schema_name);
-            match schema_xact.execute(&sql_create_schema, &[]) {
-                Ok(_) => {}
-                Err(err) => {
-                    if let Some(db_error) = err.as_db() {
-                        if db_error.code == postgres::error::UNIQUE_VIOLATION {
-                            debug!(
-                                "This is a concurrency bug with schema creation - you can \
-                                            ignore it"
-                            )
-                        }
-                    }
-                    return Err(Error::SchemaCreate(err));
-                }
-            }
-            let set_search_path = format!("SET search_path TO {}", schema_name);
-            schema_xact.execute(&set_search_path, &[]).map_err(
-                Error::SchemaSwitch,
-            )?;
-            schema_xact
-                .execute(
-                    r#"CREATE TABLE IF NOT EXISTS builder_db_migrations (
-                prefix text NOT NULL,
-                sequence_number bigint NOT NULL,
-                created_at timestamptz DEFAULT now(),
-                updated_at timestamptz,
-                PRIMARY KEY(prefix, sequence_number)
-            )"#,
-                    &[],
-                )
-                .map_err(Error::MigrationTable)?;
-            schema_xact.execute(r#"CREATE OR REPLACE FUNCTION migration_has_run_v1(p text, sn bigint) RETURNS bool AS $$
-            DECLARE
-                result BOOLEAN;
-            BEGIN
-                SELECT true FROM builder_db_migrations WHERE prefix = p AND sequence_number = sn INTO result;
-                RETURN result;
-            END
-            $$ LANGUAGE plpgsql STABLE
-            "#,
-                         &[])
-                .map_err(Error::FunctionCreate)?;
-            let next_id_v1 = format!(r#"CREATE OR REPLACE FUNCTION next_id_v1(sequence_id regclass, OUT result bigint) AS $$
+use error::Result;
+use diesel::query_dsl::RunQueryDsl;
+use diesel::sql_query;
+use diesel::pg::PgConnection;
+
+pub fn shard_setup(conn: &PgConnection, shard_id: &u32) -> Result<()> {
```

Comment:
No need to ever take a reference to a primitive type (ie. `&u32`). They're `Copy` so you just pass them by value